### ISSUE TITLE
[Camel] Kura XML route consuming from in-memory "kura-cloud" endpoint can't be restarted

### ISSUE THREAD

**Issue by hekonsek** [eclipse/kura#129]

**Comment by hekonsek** [comment]
I think I have found something. I can see this via jstack:

```
"Component Resolve Thread (Bundle 6)" #84 daemon prio=10 os_prio=0 tid=0x024e9400 nid=0xf37 waiting on condition [0x50bc4000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x5c6c5238> (a org.apache.camel.util.concurrent.RejectableFutureTask)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:426)
        at java.util.concurrent.FutureTask.get(FutureTask.java:204)
        at org.apache.camel.impl.DefaultShutdownStrategy.doShutdown(DefaultShutdownStrategy.java:197)
        at org.apache.camel.impl.DefaultShutdownStrategy.shutdown(DefaultShutdownStrategy.java:153)
        at org.apache.camel.impl.DefaultShutdownStrategy.shutdown(DefaultShutdownStrategy.java:140)
        at org.apache.camel.impl.DefaultCamelContext.stopRoute(DefaultCamelContext.java:1019)
        - locked <0x5fc40450> (a org.apache.camel.core.osgi.OsgiDefaultCamelContext)
        at org.apache.camel.impl.DefaultCamelContext.removeRouteDefinition(DefaultCamelContext.java:914)
        - locked <0x5fc40450> (a org.apache.camel.core.osgi.OsgiDefaultCamelContext)
        at org.apache.camel.impl.DefaultCamelContext.addRouteDefinitions(DefaultCamelContext.java:874)
        - locked <0x5fc40450> (a org.apache.camel.core.osgi.OsgiDefaultCamelContext)
        at org.eclipse.kura.camel.router.CamelRouter.refreshCamelRouteXml(CamelRouter.java:106)
        at org.eclipse.kura.camel.router.CamelRouter.updated(CamelRouter.java:96)
        at org.eclipse.kura.camel.router.CamelRouter.modified(CamelRouter.java:140)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.eclipse.equinox.internal.ds.model.ServiceComponent.modified(ServiceComponent.java:310)
        at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.modify(ServiceComponentProp.java:187)
        at org.eclipse.equinox.internal.ds.InstanceProcess.modifyComponent(InstanceProcess.java:690)
        at org.eclipse.equinox.internal.ds.SCRManager.processConfigurationChange(SCRManager.java:523)
        at org.eclipse.equinox.internal.ds.SCRManager.processConfigurationEvent(SCRManager.java:352)
        at org.eclipse.equinox.internal.ds.SCRManager.processConfigurationEvent(SCRManager.java:303)
        at org.eclipse.equinox.internal.ds.WorkThread.run(WorkThread.java:91)
        at org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor.run(Executor.java:70)
```

This thread blocks other threads from updating the XML route.

**Comment by hekonsek** [comment]
We should try to use VM component instead of SEDA before proceeding.

**Comment by hekonsek** [comment]
Resolved by #229. It was deadlock caused by Camel CloudClient closing in-memory queue routes synchronously.
